src/client/opamClient.ml

```diff
@@ -1449,7 +1451,7 @@ let reinstall_t t ?ask ?(force=false) ~assume_built atoms =
   let t, full_orphans, orphan_versions = check_conflicts t atoms in
 
   let requested =
-    OpamPackage.Name.Set.of_list (List.rev_map fst atoms) in
```

Comment:
`rev_map` was only here because we don't care about order, no need to `List.rev` we are converting to a set anyway :)